File: demo/src/main/java/com/gerp83/slimadaptertest/TestAdapter.java
package com.gerp83.slimadaptertest;

import android.content.Context;
import com.gerp83.slimadapter.SlimAdapter;
import java.util.ArrayList;

/**
 * Created by gerp83
 */

public class TestAdapter extends SlimAdapter{

    public TestAdapter(Context context, ArrayList<Integer> layoutIds, ArrayList<Class> classes) {
        super(context, layoutIds, classes);
    }

    @Override
    public int getItemViewTypeFromObject(int position) {
        if(position % 2 != 0) {
            return 1;

        } else if(position % 3 != 0) {
            return 2;

        }
        return 0;
    }

}

File: demo/src/main/java/com/gerp83/slimadaptertest/TestAdapter.kt
package com.gerp83.slimadaptertest

import com.gerp83.slimadapter.SlimAdapter
import java.util.ArrayList

class TestAdapter(layoutIds: ArrayList<Int>, classes: ArrayList<Class<*>>): SlimAdapter(layoutIds, classes) {

    override fun getItemViewTypeFromObject(position: Int): Int {
        if (position % 2 != 0) {
            return 1

        } else if (position % 3 != 0) {
            return 2

        }
        return 0
    }
}